test/Distributed/Distributed.Client/Scenarios/PingScenario.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Distributed.GrainInterfaces;
+using Distributed.GrainInterfaces.Ping;
+using Microsoft.Extensions.Logging;
+using Orleans;
+
+namespace Distributed.Client.Scenarios
+{
+    public class PingScenario : IScenario<PingScenario.Parameters>
+    {
+        public string Name => "ping";
+
+        private IClusterClient _clusterClient;
+        private List<(IPingRouterGrain Router, Guid[] Targets)> _requestList;
+
+        public List<Option> Options => new()
+        {
+            OptionHelper.CreateOption<int>("--routers", defaultValue: 1000),
+            OptionHelper.CreateOption<int>("--targetsPerRouter", defaultValue: 10),
+            OptionHelper.CreateOption<bool>("--doWarmup", defaultValue: true),
+        };
+
+        public async Task Initialize(IClusterClient clusterClient, Parameters p, ILogger logger)
+        {
+            _clusterClient = clusterClient;
+            _requestList = new List<(IPingRouterGrain Router, Guid[] Targets)>(p.Routers);
+
+            for (var i = 0; i < p.Routers; i++)
+            {
+                var targetIds = new Guid[p.TargetsPerRouter];
+                for (var j = 0; j < p.TargetsPerRouter; j++)
+                {
+                    targetIds[j] = Guid.NewGuid();
+                }
+
+                var routerGrain = _clusterClient.GetGrain<IPingRouterGrain>(Guid.NewGuid());
+
+                if (p.DoWarmup)
+                {
+                    try
+                    {
+                        await routerGrain.Ping(targetIds);
+                    }
+                    catch (Exception ex)
+                    {
+                        logger.LogError(ex, "Error during warmup. Ignoring.");
+                    }
+                }
+
+                _requestList.Add((routerGrain, targetIds));
+            }
+        }
+
+        public async Task IssueRequest(int request)
+        {
+            var (router, targets) = _requestList[request % _requestList.Count];
+            await router.Ping(targets);
```

Comment:
For benchmarking purposes, it may make sense to not use 'await' and to elide these instead - but we can do that test optimization later